engine-tests/src/test_utils/mod.rs

```diff
@@ -112,29 +113,28 @@ impl<'a> OneShotAuroraRunner<'a> {
         method_name: &str,
         caller_account_id: &str,
         input: Vec<u8>,
-    ) -> (Option<VMOutcome>, Option<VMError>, ExecutionProfile) {
-        let (outcome, error) = self.call(method_name, caller_account_id, input);
-        let profile = outcome
-            .as_ref()
-            .map(ExecutionProfile::new)
-            .unwrap_or_default();
-        (outcome, error, profile)
+    ) -> anyhow::Result<(VMOutcome, ExecutionProfile)> {
+        self.call(method_name, caller_account_id, input)
+            .map(|outcome| {
+                let profile = ExecutionProfile::new(&outcome);
+                (outcome, profile)
+            })
     }
 
     pub fn call(
         mut self,
         method_name: &str,
         caller_account_id: &str,
         input: Vec<u8>,
-    ) -> (Option<VMOutcome>, Option<VMError>) {
+    ) -> anyhow::Result<VMOutcome> {
         AuroraRunner::update_context(
             &mut self.context,
             caller_account_id,
             caller_account_id,
             input,
         );
 
-        match near_vm_runner::run(
+        let outcome = near_vm_runner::run(
```

Comment:
I'm curious how this is able to correctly pass on all the errors. It seems like it is working via `anyhow` magic, but I don't understand how it works. There is a new `aborted` field in [VMOutcome](https://docs.rs/near-vm-logic/0.16.1/near_vm_logic/struct.VMOutcome.html) which captures [FunctionCallError](https://docs.rs/near-vm-errors/0.16.1/near_vm_errors/enum.FunctionCallError.html)s like panics in our contract (which would be raised by not enough intrinsic gas for example). How does `anyhow` know that this field being set means that it should convert to the error variant?

Reply:
1. `VMRunnerError` derives `thiserror::Error`, which makes `VMRunnerError` suitable to use in `anyhow::Rusult`.
2. There is no magic, by the way. We just use the `aborted` field for checking errors. Like [this](https://github.com/aurora-is-near/aurora-engine/blob/chore/aleksuss/update_tests/engine-tests/src/tests/ghsa_3p69_m8gg_fwmf.rs#L46).
3. We just convert anyhow::Result<VMOutcome> into anyhow::Result<VMOutcome, EngineError> [here](https://github.com/aurora-is-near/aurora-engine/blob/chore/aleksuss/update_tests/engine-tests/src/test_utils/mod.rs#L235).

But obviously, the situation should be improved, and I have some thoughts about that.

Reply:
Ok, no magic involved, we need to inspect the `field` directly. I think we should do that directly in this function instead of requiring it to be checked at the call-site. Something like

```rust
let outcome = near_vm_runner::run(...)?;
if let Some(error) = outcome.aborted {
    return Err(error.into());
}
outcome
```

Reply:
If `outcome.aborted` is `Some`, the standalone won't process a transaction, but it must. I guess.

Reply:
The standalone should not process transactions which are aborted. In production, the Borealis Engine skips all failed Near transactions because they have no impact on the Aurora Engine state.

Reply:
I mean *try to process* and of course, we'll get the `EngineError` as a result of standalone execution if the `FunctionCallError` is some. But obviously, it's better to try to convert the `FunctionCallError` into `EngineError` and return it before the standalone execution.

P.S. I know that It looks like a cheating approach for getting `EngineError` without converting `FunctionCallError` 😄